interview-questions.md

```diff
@@ -7,81 +7,83 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: A higher order function is a function that either takes a function as an argument or returns a function when finished.
 
-  Researched answer:
+  Researched answer: Hey, I remembered the first sentence of that syllabus section almost verbatim. Go me.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JSX is JavaScript Xomething. (Probably not Xtreme. Maybe XML?) It is a combination of JavaScript and XML (which looks a lot like HTML). It allows React to inject and execute JavaScript code in the layout of a website.
 
-  Researched answer:
+  Researched answer: According to w3schools JSX stands for JavaScript XML. According to our syllabus JSX is JavaScript flavored HTML and the building block of React Components.
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a JavaScript library that is used to create highly responsive and dynamic user interfaces.
 
-  Researched answer:
+  Researched answer: Should have added that it's component based. 
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: yarn is a program created by facebook that facilitates building, testing, and debugging React websites.
 
-  Researched answer:
+  Researched answer: yarn is a package manager for code. Running yarn start creates in the current directory the file yarn.lock and the folder node_modules which contains the file .yarn-integrity.
 
 
 
 7. What is localhost:3000? How do we use localhost:3000 in the development process?
 
-  Your answer:
+  Your answer: localhost:3000 is a port on the user's host computer that can display in progress work on a website in the user's browser.
 
-  Researched answer:
+  Researched answer: localhost:3000 is the default port on the user's computer used by React and rails to demo your project while you're working on it.
 
 
 
 8. What is an iterable in JavaScript? What data types are iterables? How can this information useful to you as a developer?
 
-  Your answer:
+  Your answer: Probably something to do with loops since they iterate.
 
-  Researched answer:
+  Researched answer: An iterable in JavaScript is an object that can be iterated over, i.e. it has a series of elements that can be accessed and interacted with sequentially. Strings and arrays are iterable. This property is valuable as it allows to utilize operations (loops, etc.) and and built-in methods (map, filter, etc.) on the iterables.
 
 
 
 9. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: I have seen this word. Related to React. That's all the context my brain can currently give me. Maybe it's importing / exporting classes and methods between parents and children?
 
-  Researched answer:
+  Researched answer: I was thinking of state hoisting which is accomplished by passing a callback from a container component to a child component. (I don't fully understand what those words mean.)
```

Comment:
Additionally: Hoisting is BAD PRACTICE. We don't want to be declaring our variables and such after the functions that utilize them. JS does us some favors by hoisting, but we should not ever rely on it.